app/src/main/java/ru/practicum/android/diploma/filters/presentation/ui/ChooseIndustry.kt

```diff
@@ -30,41 +50,152 @@ class ChooseIndustry : Fragment() {
-                ChooseIndustryResult.NoInternet -> println("no internet")
-                is ChooseIndustryResult.Success -> showIndustry(state.response)
+                ChooseIndustryResult.EmptyResult -> showInternetError()
+                is ChooseIndustryResult.Error -> showInternetError()
+                ChooseIndustryResult.NoInternet -> showInternetError()
+                is ChooseIndustryResult.Success -> {
+                    hideInternetError()
+                    showIndustry(state.response)
+                }
             }
         }
     }
 
     private fun initIndustryAdapter(): IndustryAdapter {
-        industryAdapter = IndustryAdapter {
+        val industryText = arguments?.getString("INDUSTRY_TEXT")
+        industryRecyclerView = binding.industryRecycler
+        industryRecyclerView.layoutManager = LinearLayoutManager(requireContext())
 
+        industryAdapter = IndustryAdapter(
+            industryList,
+            { industry ->
+                selectedIndustry = industry
+            },
+            industryText
+        )
+
+        industryRecyclerView.adapter = industryAdapter
+        return industryRecyclerView.adapter as IndustryAdapter
+    }
+
+    private fun debounce() {
+        debounce = debounce<String>(
+            SEARCH_DEBOUNCE_DELAY_MILLIS,
+            viewLifecycleOwner.lifecycleScope,
+            true,
+            action = { text ->
+                if (text.isNullOrEmpty()) {
+                    hideErrorNothing()
+                    industryList.clear()
+                    industryListFilter.clear()
+                    industryList.addAll(savedIndustryList)
+                    industryAdapter.notifyDataSetChanged()
+                } else {
+                    industryList.clear()
+                    industryListFilter.clear()
+                    savedIndustryList.forEach {
+                        if (it.name.lowercase() == text.lowercase()) {
+                            industryListFilter.add(it)
+                        }
+                    }
+                    if (industryListFilter.isEmpty()) {
+                        showErrorNothing()
+                    } else {
+                        hideErrorNothing()
+                    }
+                    industryList.addAll(industryListFilter)
+                    industryAdapter.notifyDataSetChanged()
+                }
+            })
+    }
+
+    private fun initListeners() {
+        binding.selectBtn.visibility = View.GONE
+        binding.btnBack.setOnClickListener {
+            findNavController().navigateUp()
+        }
+        industryAdapter.itemClickListener = { position, vacancy ->
+            industryAdapter.notifyDataSetChanged()
+            binding.selectBtn.visibility = View.VISIBLE
+        }
+        binding.selectBtn.setOnClickListener {
+            if (selectedIndustry == null) {
+                return@setOnClickListener
+            }
+            btSelectClicked(selectedIndustry!!)
```

Comment:
🍏 Можно записать проще:
```suggestion
            selectedIndustry?.let {
                 btSelectClicked(it)
             }
```

https://kotlinlang.org/docs/scope-functions.html